pkg/kube/resource.go

```diff
@@ -29,6 +29,9 @@ func (r *ResourceList) Append(val *resource.Info) {
 // Visit implements resource.Visitor. The visitor stops if fn returns an error.
 func (r ResourceList) Visit(fn resource.VisitorFunc) error {
 	for _, i := range r {
+		if i.Name == "" {
```

Comment:
This function is a generic visitor for any/all scenarios. It should not implement business logic like skipping objects missing a name